packages/block-library/src/cover/edit/index.js

```diff
@@ -112,6 +112,10 @@ function CoverEdit( {
 	);
 	const mediaUrl = media?.source_url;
 
+	if ( mediaUrl && useFeaturedImage ) {
+		setCoverIsDark( mediaUrl, dimRatio, overlayColor.color );
+	}
```

Comment:
Will this be called on every render? Would it be worth only calling if `mediaUrl`, `dimRatio`, or `overlayColor.color` changes?

That might wind up re-introducing some form of `useEffect`, though, which might not be the goal here 🤔

Reply:
Yeh, this[ may be one place we need an effect still.](https://github.com/WordPress/gutenberg/pull/53253#discussion_r1282628098)